### ISSUE TITLE
Allow MFS to be mounted to the File System

### ISSUE THREAD

**Issue by hacdias** [ipfs-shipyard/ipfs-desktop#618]
Should IPFS Desktop allow the user to mount MFS to the file system or should this feature be incorporated into js-ipfs and go-ipfs?

Ref.: https://github.com/tableflip/ipfs-fuse

/cc @ipfs-shipyard/ipfs-gui-team

**Comment by lidel** [comment]
I feel it is worth highlighting that @alanshaw started work towards that goal:

- https://github.com/tableflip/ipfs-fuse
  - Demo:  "Mount IPFS MFS as a FUSE volume" https://youtu.be/jXkTEBdM6aA

**Comment by hacdias** [comment]
@lidel yeah, I was thinking about using that package to enable/disable the mounting.

**Comment by lidel** [comment]
See also: https://github.com/piedar/js-ipfs-mount

**Comment by hacdias** [comment]
Uhhh, that's cool. I'll try and see if it works on Windows with Dokany

**Comment by djdv** [comment]
@lidel 
>mounting MFS will be a part of go-ipfs

At this time, yes!

>Do you have any ideas on how we could integrate it with ipfs-desktop?

Since we're just exposing it as a writable filesystem, you should be able to interact with it in a standard way.
I haven't nailed everything down yet, but in theory you should be able to parse the config to find the mountpoint, and add some wrapping around it.

You can see the status of the mount implimentation here:
https://github.com/ipfs/go-ipfs/issues/5003
https://www.youtube.com/playlist?list=PLrZxGPqQuwHC_cZF2assuM0Vl3D5s5OSN

**Comment by hacdias** [comment]
@djdv just read the whole issue and I have to say that what I read is just amazing and the progress you've made is incredible. I can't wait to be able to easily mount MFS on Windows.

Just as a side note: I was wondering if it would be possible to have an IPFS implementation independent mount that would work with any IPFS implementation - via HTTP API or connecting directly maybe. The drawback I see here is that the connection might be a bit slower. Not sure. Needs thought.

**Comment by hacdias** [comment]
@djdv what's the state of MFS mount?

**Comment by djdv** [comment]
@hacdias 
Current status is being tracked here:
https://github.com/ipfs/package-managers/issues/74

At the moment I'm making an effort to develop versions of the concepts demoed before, in a way that can be merged into `go-ipfs` gradually. I'd like to have some kind of read support under an experimental flag ASAP.

I wrote a big summary [here](https://github.com/ipfs/package-managers/issues/74#issuecomment-519122152) but the tl;dr is that above ^

**Comment by bam80** [comment]
> Current status is being tracked here:
> [ipfs-inactive/package-managers#74](https://github.com/ipfs-inactive/package-managers/issues/74)

That repo has been archived:
> I apologize to the community, to which I made promises I could not keep.

So what's next? Were these efforts continued, and where to track the status now?

**Comment by pirate** [comment]
I know it's just a drop in the bucket, but I'd be willing to chip in $150 into any kind of gittip/bounty/donation system towards this feature. A mutable FUSE mount for IPFS one of the biggest things I'm waiting for that would allow us to easily integrate IPFS everywhere in our internet archiving stack over at ArchiveBox.io.

**Comment by lidel** [comment]
IIUC FUSE turned out to be [pretty problematic part of go-ipfs](https://github.com/ipfs/go-ipfs/issues?page=2&q=is%3Aissue+is%3Aopen+fuse),  each platform/filesystem (linux, windows, macos) having own issues, making the entire thing too brittle for production use, and fixing myriad of fuse problems was stealing dev time and burning our devs in the process.  It is unlikely to be fixed any time soon upstream.

That being said, if someone is brave enough to tackle this across all three platforms, PRs welcome.
MFS mounting for desktop use does not need to be super performant nor a part of go-ipfs: ipfs-desktop is an electron app, so if someone can make it work in JS (perhaps by reusing prior art from [Alan's demo](https://www.youtube.com/watch?v=jXkTEBdM6aA)), I'd be happy to help with review. 

Personally, I'd like to see someone experiment with exposing MFS over WebDAV instead, which is HTTP-based and seems to be supported on [Linux, macOS](https://help.dreamhost.com/hc/en-us/articles/216473527-Accessing-WebDAV-with-Mac-OS-X-and-Linux) and [Windows](https://www.youtube.com/watch?v=i5ubScGwUOc) out of the box (in form of "mounting external drives").

**Comment by pirate** [comment]
I have no particular love for FUSE either, NFS/SMB/WebDAV, any mountable mutable filesystem works for me.
